src/main/java/juja/sqlcmd/DatabaseManager.java

```diff
@@ -30,4 +43,21 @@ public boolean connect(String database, String user, String password) {
             return false;
         }
     }
+
+    public static void main(String[] args) throws SQLException {
+        DatabaseManager manager = new DatabaseManager();
+        manager.connect("sqlcmd", "sqlcmd", "sqlcmd");
+        System.out.println(Arrays.toString(manager.getTableNames()));
+    }
+
+    public String[] getTableNames() throws SQLException {
+        try (ResultSet rs = connection.getMetaData().getTables(null, schemaPattern, "%", new String[]{"TABLE"})) {
+            List<String> tablesList = new LinkedList<>();
+            while (rs.next()) {
+                tablesList.add(rs.getString("table_name"));
+            }
+            return tablesList.toArray(new String[0]);
+        }
```

Comment:
I'm not sure that such an implementation getTableNames() corresponds to the requirement of the task.

Reply:
Yes, I know it’s cheating, I wrote about it in Slack, but anyway, my hands couldn’t write rough code with arrays..))

Reply:
@AVM82 @danilkuznetsov sorry I really don't see any reasons for arrays. Metadata is a default implementation for retrieving tables from databases. It's database independent. But this implementation does not imply neither Count(*) nor ResultSet.last methods for creating new Array[size].

Reply:
Using next trick you can get a size of your ResultSet: